README.md

```diff
@@ -75,38 +84,36 @@ Share Social , Sending Simple Data to Other Apps
         </provider>
       </application>
     ```
-
+    
   - Create a `filepaths.xml` under this directory:
-    `android/app/src/main/res/xml`. In this file, add the following contents:
-
+    `android/app/src/main/res/xml`. 
+    
+    In this file, add the following contents:
+    
+    File: `android/app/src/main/res/filepaths.xml`
     ```xml
       <?xml version="1.0" encoding="utf-8"?>
       <paths xmlns:android="http://schemas.android.com/apk/res/android">
           <external-path name="myexternalimages" path="Download/" />
       </paths>
     ```
-
-6. Edit your `MainApplication` class to implement `ShareApplication`
-  - Also add the `getFileProviderAuthority` method to your MainApplication class,
-    and have it return the `android:authorities` that was added in
-    AndroidManifest file.
-  - For example: Replace the `com.example.yourappidhere` below with the
-    `applicationId` that is defined in your `android/app/build.gradle`. It must
-    be [hard-coded here to work
-    properly](https://github.com/EstebanFuentealba/react-native-share/issues/200#issuecomment-361938532).
+    
+6. Edit your `MainApplication.java` class to add `implements ShareApplication` and `getFileProviderAuthority`
+  - The `getFileProviderAuthority` function returns the `android:authorities` value added in the `AndroidManifest.xml` file
+  - `applicationId` is defined in the `defaultConfig` section of your `android/app/build.gradle` and referenced using `BuildConfig.APPLICATION_ID`
 
     ```
     import cl.json.ShareApplication
-
-    class MyApplication extends Application implements ShareApplication, ReactApplication {
+    public class MainApplication extends Application implements ShareApplication, ReactApplication {
+    class MainApplication extends Application implements ShareApplication, ReactApplication {
 
     {
 
          //...
 
          @Override
          public String getFileProviderAuthority() {
-                return "com.example.yourappidhere.provider";
+                return BuildConfig.APPLICATION_ID + ".provider";
```

Comment:
just curious, why `BuildConfig`?

Reply:
My goal was to avoid duplication in source code. I did not like hardcoding "com.example.yourappidhere.provider" string in *both* AndroidManifest.xml and MainApplication. getFileProviderAuthority().  

However, your comment made me realize I still have duplication at:
1) android/app/build.gradle 
2) Also declared in the AndroidManifest.xml manifest/package 

````
<manifest 
xmlns:android="http://schemas.android.com/apk/res/android"
package="com.example.yourappidhere" 
android:versionCode="1"
android:versionName="1.0">
````

The problem is I could not find a solution for AndroidManifest.xml to self-reference the manifest/package attribute from the provider/android:authorities in the AndroidManifest.xml  ?

My solution was to use a variable ${applicationId} in AndroidManifest which required declaraion in build.gradle.
````
<provider 
android:name="android.support.v4.content.FileProvider" android:authorities="${applicationId}.provider"0
android:exported="false" 
android:grantUriPermissions="true">
			<meta-data 
android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/provider_paths" />
</provider>
````

This might achieve my goal of no duplication
````
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="${applicationId}"
````

Is Context.getPackageName() preferable to  BuildConfig.APPLICATION_ID ?
````
  @Override
  public String getFileProviderAuthority() {
    // return BuildConfig.APPLICATION_ID + ".provider";
    return Context.getPackageName() + ".provider";
  }
````

Or perhaps this doc change could create problems for other users.  

Perhaps I should just do what works for myself and leave docs as is and let users determine their own preferred solution?

Reply:
I am no Android expert.  

There may be a difference between Android applicationid versus manifest package that I do not understand.

Reply:
Thanks for the response. Solid read.